refinery/data_set_manager/views.py

```diff
@@ -1059,11 +1060,20 @@ def post(self, request):
             return HttpResponseForbidden()
 
         logger.debug("Adding files to data set '%s'", data_set)
-        for file_store_item in data_set.get_file_store_items():
-            if (file_store_item.source.startswith(
+        node = get_object_or_404(Node, uuid=request.data.get('node_uuid'))
+        file_store_item = node.get_file_store_item()
+        if (file_store_item and not file_store_item.datafile and
+                file_store_item.source.startswith(
                     (settings.REFINERY_DATA_IMPORT_DIR, 's3://')
-            )):
-                import_file.delay(file_store_item.uuid)
+                )):
+            # Remove the FileStoreItem's import_task_id to treat it as a
+            # brand new import_file task when called below.
+            # We then have to update its Node's Solr index entry, so the
+            # updated file import status is available in the UI.
+            file_store_item.import_task_id = ""
```

Comment:
So I attempted to update the state of the existing `import_file` task like so:
```
import_file.update_state(
    task_id=file_store_item.import_task_id,
    state=celery.states.PENDING,
    meta="FileBrowser UI data file addition"
)
```

Which also allows for the `PENDING` state to get to the UI (and is more clear as to what is going on), but due to [some logic within `import_file`](https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/file_store/tasks.py#L166-L176) the `import_file` task below this will fail because the `FileStoreItem` still has an `import_task_id`, and its corresponding task is in an `UNREADY_STATE` (`PENDING`). 

Updating the state of an existing import_file task felt invasive in the sense that I had to update the `import_file` signature to allow for passing in a notion of whether this importing file was a special case (User adding a datafile from the FileBrowser UI) or not to avoid hitting the logic that would fail the task.

A `PENDING` state will be assigned to a file `Node` if its corresponding `FileStoreItem` has no download url and its import status isn't in a `READY_STATE`

Reply:
Have you considered updating the logic in `_get_download_url_or_import_state()` to support this new use case properly?

Reply:
@hackdna I don't believe the logic in `_get_download_url_or_import_state()` is incorrect, or in need of updating to handle this use case? Could you elaborate?